path_role.go

```diff
@@ -0,0 +1,346 @@
+				logical.UpdateOperation: b.pathRoleCreateUpdate,
+				logical.ReadOperation:   b.pathRoleRead,
+				logical.DeleteOperation: b.pathRoleDelete,
+			},
+			HelpSynopsis:    strings.TrimSpace(roleHelp["role"][0]),
+			HelpDescription: strings.TrimSpace(roleHelp["role"][1]),
+		},
+	}
+}
+
+type jwtRole struct {
+	// Policies that are to be required by the token to access this role
+	Policies []string `json:"policies"`
+
+	// TokenNumUses defines the number of allowed uses of the token issued
+	NumUses int `json:"num_uses"`
+
+	// Duration before which an issued token must be renewed
+	TTL time.Duration `json:"ttl"`
+
+	// Duration after which an issued token should not be allowed to be renewed
+	MaxTTL time.Duration `json:"max_ttl"`
+
+	// Period, if set, indicates that the token generated using this role
+	// should never expire. The token should be renewed within the duration
+	// specified by this value. The renewal duration will be fixed if the
+	// value is not modified on the role. If the `Period` in the role is modified,
+	// a token will pick up the new value during its next renewal.
+	Period time.Duration `json:"period"`
+
+	// Role binding properties
+	BoundAudiences []string                      `json:"bound_audiences"`
+	BoundSubject   string                        `json:"bound_subject"`
+	BoundCIDRs     []*sockaddr.SockAddrMarshaler `json:"bound_cidrs"`
+	UserClaim      string                        `json:"user_claim"`
+	GroupsClaim    string                        `json:"groups_claim"`
+}
+
+// role takes a storage backend and the name and returns the role's storage
+// entryÍ
+func (b *jwtAuthBackend) role(ctx context.Context, s logical.Storage, name string) (*jwtRole, error) {
+	raw, err := s.Get(ctx, "role/"+strings.ToLower(name))
+	if err != nil {
+		return nil, err
+	}
+	if raw == nil {
+		return nil, nil
+	}
+
+	role := new(jwtRole)
+	if err := json.Unmarshal(raw.Value, role); err != nil {
+		return nil, err
+	}
+
+	return role, nil
+}
+
+// pathRoleExistenceCheck returns whether the role with the given name exists or not.
+func (b *jwtAuthBackend) pathRoleExistenceCheck(ctx context.Context, req *logical.Request, data *framework.FieldData) (bool, error) {
+	role, err := b.role(ctx, req.Storage, data.Get("name").(string))
+	if err != nil {
+		return false, err
+	}
+	return role != nil, nil
+}
+
+// pathRoleList is used to list all the Roles registered with the backend.
+func (b *jwtAuthBackend) pathRoleList(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roles, err := req.Storage.List(ctx, "role/")
+	if err != nil {
+		return nil, err
+	}
+	return logical.ListResponse(roles), nil
+}
+
+// pathRoleRead grabs a read lock and reads the options set on the role from the storage
+func (b *jwtAuthBackend) pathRoleRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName := data.Get("name").(string)
+	if roleName == "" {
```

Comment:
I don't think this check (present in the operations, too), is necessary. "name" is a url parameter so the framework won't even let the request get here if it's missing.